platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/ConfigurationCacheState.kt

```diff
@@ -444,13 +441,13 @@ class ConfigurationCacheState(
     }
 
     private
-    suspend fun WriteContext.writeBuildWithNoWork(state: BuildState, rootBuild: VintageGradleBuild) {
-        withGradleIsolate(rootBuild.gradle, userTypesCodec) {
-            writeString(state.identityPath.path)
-            if (state.isProjectsCreated) {
+    suspend fun WriteContext.writeBuildWithNoWork(buildState: BuildState) {
+        withGradleIsolate(buildState.mutableModel, userTypesCodec) {
```

Comment:
🤔  Technically, there is a small semantic change here, but I don't understand why the code was like it was. We are writing state of some build (not necessarily root), which we know has no work. But we use the root build's `Gradle` is the isolate. This seems like an inconsistency.

My guess is that it does not actually matter, because everything we serialize are just primitives and one data class. So the change of the isolate has no effective consequences.

It seems more correct, though, to use the `Gradle` of the target build as the isolate